src/Editor/GUI/Editors/CurveEditor.cpp

```diff
@@ -0,0 +1,740 @@
+            "Quart in",
+            "Quart out",
+            "Quart in  out",
+            "Quint in",
+            "Quint out",
+            "Quint in  out",
+            "Sine in",
+            "Sine out",
+            "Sine in  out",
+            "Expo in",
+            "Expo out",
+            "Expo in  out",
+            "Circ in",
+            "Circ out",
+            "Circ in  out",
+            "Elastic in",
+            "Elastic out",
+            "Elastic in  out",
+            "Back in",
+            "Back out",
+            "Back in  out",
+            "Bounce in",
+            "Bounce out",
+            "Bounce in out",
+
+            "Sine square",
+            "Exponential",
+
+            "Schubring1",
+            "Schubring2",
+            "Schubring3",
+
+            "SinPi2",
+            "Swing"
+        };
+        if (modified) {
+            item = 0;
+        }
+        if (ImGui::Combo("Ease type", &item, items, IM_ARRAYSIZE(items))) {
+            max = maxpoints;
+            if (item > 0) {
+                for (i = 0; i < max; ++i) {
+                    points[i].x = i / float(max - 1);
+                    points[i].y = float(tween::ease(item - 1, points[i].x));
+                }
+            }
+        }
+
+        char buf[128];
+        const char *str = label;
+
+        if (hovered) {
+            ImVec2 pos = (g.IO.MousePos - bb.Min) / (bb.Max - bb.Min);
+            pos.y = 1 - pos.y;
+
+            sprintf(buf, "%s (%f,%f)", label, pos.x, pos.y);
+            str = buf;
+        }
+
+        RenderTextClipped(ImVec2(bb.Min.x, bb.Min.y + style.FramePadding.y), bb.Max, str, NULL, NULL, ImVec2(100, 100));
+
+        return modified;
+    }
+
+};
+
+CurveEditor::CurveEditor() {
+    curves.clear();
+}
+
+CurveEditor::~CurveEditor() {
+    curves.clear();
+}
+
+void CurveEditor::Show() {
+
+    UpdateCurves(0.1f, 100.0f);
+
+    if (ImGui::Begin("Curve editor")) {
+        ImGui::InputText("Unique curve name", curveBuf, 10);
+        if (ImGui::Button("Add curve +")) {
+            AddMyCurve(curvename, addedCurve, 0);
+            addedCurve++;
+        }
+
+        RenderCurveEditor();
+    }
+
+    ImGui::End();
+
+}
+
+bool CurveEditor::IsVisible() const {
+    return visible;
+}
+
+void CurveEditor::SetVisible(bool visible) {
+    this->visible = visible;
+}
+
+void CurveEditor::AddMyCurve(std::string& curve_name, ImGuiID uniqueId, int item) {
+    MyCurve curve;
+    curve.curve_name = curve_name;
+    curve.value[0].x = -1;
+    curve.id = uniqueId;
+    curve.item = item;
+
+    curve.editVelocityX = false;
+    curves.push_back(curve);
+}
+
+void CurveEditor::UpdateCurves(float deltaTime, float totalTime) {
+
+    if (time >= totalTime) {
+        time = totalTime;
+        goBack = true;
+    }
+    if (time <= 0.0f) {
+        time = 0.0f;
+        goBack = false;
+    }
+    
+    if (!goBack)
+        time += deltaTime;
+
+    if (goBack)
+        time -= deltaTime;
+
+    float tempValue;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The scope of the variable 'tempValue' can be reduced. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AvariableScope)